optax/contrib/_ademamix.py

```diff
@@ -0,0 +1,220 @@
+"""AdEMAMix.
+
+Implementation of
+"THE ADEMAMIX OPTIMIZER: BETTER, FASTER, OLDER"
+(https://arxiv.org/pdf/2409.03137) by Matteo Pagliardini, 
+Pierre Ablin and David Grangier.
+"""
+
+from typing import NamedTuple, Tuple
+import chex
+import jax.numpy as jnp
+import jax.tree_util as jtu
+from optax._src import base
+from optax._src import combine
+from optax._src import numerics
+from optax._src import transform
+import optax.tree_utils as otu
+
+class ScaleByAdemamixState(NamedTuple):
+  """State for the Ademamix algorithm."""
```

Comment:
Could you add a description of the attributes in the docstring?
See e.g. https://github.com/google-deepmind/optax/blob/b8c2e133319480509576a7280f851e9d6ec7dccf/optax/_src/transform.py#L1448

Reply:
I took a stab.  LMK if there are better descriptions I could do.  I (theoretically) could add the math update if you believe that would be warranted.  I also want to get the description right.  `nu` is still the estimate of the second moment, but now the first moment is a combination of `m1` and `m2` so I just stated their respective EMA types.

Reply:
Looks great to me.
I think I would say "fast EMA of first moment" rather than first moment personally (same for "slow EMA" and this would apply in the other docstrings). But if you have stronger preferences I won't argue against.